src/ImmersedBoundaries/partial_cell_bottom.jl

```diff
@@ -95,10 +95,13 @@ Criterion is h ≥ z - ϵ Δz
 
 """
 @inline function _immersed_cell(i, j, k, underlying_grid, ib::PartialCellBottom)
-    # Face node above current cell
-    z = znode(i, j, k+1, underlying_grid, c, c, f)
-    h = @inbounds ib.bottom_height[i, j, 1]
-    return z ≤ h
+    # Face node below current cell
+    z  = znode(i, j, k, underlying_grid, c, c, f)
+    h  = @inbounds ib.bottom_height[i, j, 1]
+    ϵ  = ib.minimum_fractional_cell_height
+    # z + Δz is equal to the face above the current cell
+    Δz = Δzᶜᶜᶜ(i, j, k, ibg.underlying_grid)
+    return z + Δz * ϵ ≤ h 
```

Comment:
Should we expect this change to fix the weird behaviors we were seeing in https://github.com/CliMA/Oceananigans.jl/discussions/3529 ?

cc @francispoulin

Reply:
I am not sure, @jm-c is the author of this fix, and he said that with this fix the immersed boundary example did not NaN immediately but it still NaNed after a little bit so we believe there is more bugs to be squashed. You can try that example on this and see if it makes any difference. It will most likely change the solution.

Reply:
I am happy to try the example from #3529 it on this branch, when you think it's ready for testing.